coffee/rewriter.py

```diff
@@ -253,6 +210,179 @@ def factorize(self, mode='standard'):
         # Perform the factorization
         self.expr_factorizer.factorize(should_factorize)
 
+    def reassociate(self, reorder=lambda x: x):
+        """Reorder symbols in associative operations following a convention.
+        By default, the convention is to order the symbols based on their rank.
+        For example, the terms in the expression ::
+
+            a*b[i]*c[i][j]*d
+
+        are reordered as ::
+
+            a*d*b[i]*c[i][j]
+
+        This as achieved by reorganizing the AST of the expression.
+        """
+
+        def _reassociate(node, parent):
```

Comment:
Yes, but I wrote it before Lawrence wrote the infrastructure, so maybe one day we can turn it into a visitor. Not gonna happen now.